File: 1461-check-if-a-string-contains-all-binary-codes-of-size-k/1461-check-if-a-string-contains-all-binary-codes-of-size-k.cpp
class Solution {
public:
    bool hasAllCodes(string s, int k) {
//         set<string> ss;
//         for(int i=0;i<=s.size()-k;++i)
//         {
//             ss.insert(s.substr(i,i+k));
//         }
        
//         return (ss.size()==pow(2,k));
        
        
         set<string> strs;
        for(int i=0;i+k<=s.size();i++) 
            strs.insert(s.substr(i, k));
        return strs.size()==pow(2, k);
    }
};

File: 1461-check-if-a-string-contains-all-binary-codes-of-size-k/1461-check-if-a-string-contains-all-binary-codes-of-size-k.java
class Solution {
    public boolean hasAllCodes(String s, int k) {
          Set<String> ss=new HashSet();
        for(int i=0;i<=s.length()-k;++i)
            ss.add(s.substring(i,i+k));
        
        return (ss.size()==Math.pow(2,k));
    }
}